lib/Mojo/WebSocketProxy/Backend/ConsumerGroups.pm

```diff
@@ -427,4 +438,38 @@ sub _prepare_request_data {
     return $msg_type, $request_data;
 }
 
+=head2 get_proper_stream_name
+
+This method looking for stream name that message should be added to,
+I<general> is the default stream name if either C<$proposed_stream_name> is not provided or C<queue_separation_enabled> is 0.
+Also C<$proposed_stream_name> will be ignored if C<queue_separation_enabled> is 0.
+
+=over 4
+
+=item * C<$proposed_stream_name> - The stream/category name
+
+=back
+
+Return name of stream
+
+=cut
+
+sub get_proper_stream_name {
+    my ($self, $proposed_stream_name) = @_;
+
+    my $stream //= DEFAULT_CATEGORY_NAME;
+    $proposed_stream_name //= '';
+
+    if ($self->queue_separation_enabled) {
+        return $stream if $self->ignore_separations && any {
```

Comment:
Why do we need this?

(if we did need it for something, please use a hashref instead of an array)

Reply:
Not sure I got you but if you mean `ignore_separations`, we have to have more control over queues, and with this new mechanism the DevOps gonna be able to ignore different queues and inform the WebSocket to pass the messages through `general`. it works based on chef tags. once the card gets into QA, I'll add more detail to the documentation.

Reply:
@tom-binary @mehdi-fs this is misunderstanding, devops was asking to disable rpc workers if category is not presented in tags ... API should not have this feature. We'll remove this logic

Reply:
@kirill-binary this feature is going to be quite useful for redirect requests under a category to the default category in case of any failure and problem.

Reply:
What's the outcome on this? `queue_separation_enabled` and `ignore_sepration` looks like operation logic embedded in the code.

Reply:
We're going to keep the queue ignoring mechanism and move all operational logics to the `binary_websocket_api` mojo hooks

Reply:
@tom-binary @raunakkathuria the point of this logic is  in case if we miscount with number of workers or timeout value, that we'll have possibility to disable this category, without reverting the code.